// react-navigation-header-buttons/lib/commonjs/overflowMenu/vendor/MenuItem.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MenuItem = MenuItem;
var React = _interopRequireWildcard(require("react"));
var _reactNative = require("react-native");
var _elements = require("@react-navigation/elements");
var _native = require("@react-navigation/native");
var _jsxRuntime = require("react/jsx-runtime");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
/**
 * A component to show a single list item inside a Menu.
 */
function MenuItem(props) {
  const {
    icon,
    title,
    disabled,
    onPress,
    style,
    titleStyle,
    testID
  } = props;
  const {
    dark,
    colors: {
      text
    }
  } = (0, _native.useTheme)();
  const disabledColor = dark ? styles.darkDisabled : styles.lightDisabled;
  const titleColor = disabled ? disabledColor : {
    color: text
  };
  const themePressColorAndroid = dark ? 'rgba(255, 255, 255, .32)' : 'rgba(0, 0, 0, .32)';
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_elements.PlatformPressable, {
    style: [styles.container, style],
    onPress: onPress,
    disabled: disabled,
    testID: testID,
    pressColor: themePressColorAndroid,
    children: /*#__PURE__*/(0, _jsxRuntime.jsxs)(_reactNative.View, {
      style: styles.row,
      children: [/*#__PURE__*/React.isValidElement(icon) && /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactNative.View, {
        style: [styles.item, styles.icon],
        pointerEvents: "box-none",
        children: icon
      }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactNative.View, {
        style: [styles.item, styles.content, icon != null ? styles.widthWithIcon : undefined],
        pointerEvents: "none",
        children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactNative.Text, {
          selectable: false,
          numberOfLines: 1,
          style: [styles.title, titleColor, titleStyle],
          children: title
        })
      })]
    })
  });
}
const minWidth = 112;
const maxWidth = 280;
const iconWidth = 25;
const styles = _reactNative.StyleSheet.create({
  container: {
    paddingHorizontal: 8,
    minWidth,
    maxWidth,
    height: 48,
    justifyContent: 'center'
  },
  row: {
    flexDirection: 'row'
  },
  icon: {
    width: iconWidth
  },
  title: {
    fontSize: 16
  },
  item: {
    marginHorizontal: 8
  },
  content: {
    justifyContent: 'center',
    minWidth: minWidth - 16,
    maxWidth: maxWidth - 16
  },
  widthWithIcon: {
    maxWidth: maxWidth - (iconWidth + 48)
  },
  lightDisabled: {
    color: 'rgba(0, 0, 0, 0.32)'
  },
  darkDisabled: {
    color: 'rgba(255, 255, 255, 0.32)'
  }
});
